Sources/Sentry/SentryProfiler.mm

```diff
@@ -370,56 +422,30 @@ + (SentryEnvelopeItem *)createProfilingEnvelopeItemForTransaction:(SentryTransac
     payload[@"transactions"] = @[ transactionInfo ];
 
     // add the gathered metrics
-    NSMutableDictionary<NSString *, id> *metrics = [_gCurrentProfiler->_metricProfiler serialize];
-
-    const auto slicedMetrics =
-        [NSMutableDictionary<NSString *, id> dictionaryWithCapacity:metrics.count];
-    void (^metricSlicer)(NSString *_Nonnull, NSDictionary<NSString *, id> *_Nonnull, BOOL *_Nonnull)
-        = ^(NSString *_Nonnull metricKey, NSDictionary<NSString *, id> *_Nonnull nextMetricsEntry,
-            BOOL *_Nonnull metricStop) {
-              NSArray<NSDictionary<NSString *, NSString *> *> *nextMetrics
-                  = nextMetricsEntry[@"values"];
-
-              const auto nextSlicedMetrics = [self slicedArray:nextMetrics transaction:transaction];
-              if (!nextSlicedMetrics) {
-                  return;
-              }
-
-              slicedMetrics[metricKey] =
-                  @{ @"unit" : nextMetricsEntry[@"unit"], @"values" : nextSlicedMetrics };
-          };
-    [metrics enumerateKeysAndObjectsUsingBlock:metricSlicer];
+    const auto metrics = [_gCurrentProfiler->_metricProfiler serializeForTransaction:transaction];
```

Comment:
This implicit slicing of metrics entries now takes place in `SentryMetricProfiler::serializeValuesWithNormalizedTime(...)` (called by `-[SentryMetricProfiler serializeForTransaction:]`).